### PR TITLE
Add lock counter to single_diff_t, move check_lock_counters into libswiftnav

### PR DESCRIPTION
**Pull request by fnoble**
Supports swift-nav/piksi_firmware#434

@kovach

### CODE REVIEW

File: src/observation.c [hunk 1]
```diff
@@ -220,6 +222,45 @@ u8 make_propagated_sdiffs(u8 n_local, navigation_measurement_t *m_local,
...
+  assert(sds != NULL);
+  assert(lock_counters != NULL);
+  assert(sats_to_drop != NULL);
+
+  u8 num_sats_to_drop = 0;
+  for (u8 i = 0; i<n_sds; i++) {
+    u8 prn = sds[i].prn;
+    u16 new_count = sds[i].lock_counter;
+    if (new_count != lock_counters[prn]) {
+      sats_to_drop[num_sats_to_drop++] = prn;
```
Comment (cbeighley) [inline src/observation.c]: Do we have a style recommendation for using an increment inside an array index like this? I feel like it makes it less readable.

Comment (fnoble) [inline src/observation.c]: Its a pretty common pattern in C, but I'll check the JPL standard

Comment (cbeighley) [inline src/observation.c]:
```
Rule 24
There should be no more than one statement or variable declaration per
line. A single exception is the C for-loop, where the three controlling
expressions (initialization, loop bound, and increment) can be placed on a
single line. 
```

Comment (fnoble) [inline src/observation.c]: I don't think that applies, this is technically only one statement.

Comment (fnoble) [inline src/observation.c]: @gsmcmullin @henryhallam : you guys have the most C experience, do you find this pattern objectionable?

Comment (gsmcmullin) [inline src/observation.c]: No objection.  I think it's pretty clear what's going on.  Expanding it will just clutter up the code.

Comment (cbeighley) [inline src/observation.c]: I'd argue it's a compound statement, and two statements for the sake of JPL Rule 24; it assigns to array element and increments a variable. But I'll leave it up to you - lgtm

**Pull request merged**